util/src/main/scala/scala/scalanative/io/package.scala

```diff
@@ -1,22 +1,26 @@
 package scala.scalanative
 
-import java.nio.ByteBuffer
+import java.nio.{BufferOverflowException, ByteBuffer}
 import java.nio.file.Path
+import scala.annotation.tailrec
 import scala.collection.JavaConverters._
+import scala.util._
 
 package object io {
+  type NamedPool = (String, ByteBufferPool)
   // We allocate a pool of direct buffers. Due to virtual memory
   // semantics, only the used part is going to be actually provided
   // by the underlying OS.
-  private val pool = new ByteBufferPool
-
-  def withScratchBuffer(f: ByteBuffer => Unit): Unit = {
-    val buffer = pool.claim()
-    buffer.clear
-    try f(buffer)
-    finally pool.reclaim(buffer)
+  private val namedPools: List[NamedPool] = List(
+    "small" -> 8 * 1024 * 1024,
+    "large" -> 40 * 1024 * 1024
```

Comment:
and at some point you added `"huge"` and `"impossible"` size when someone try to compile very big and complicated file?

I do have files like, but I don't move forward to it yet :)

This files requires 60Gb heap to be compiled via scala-native, when scala and scala-js working fine on default settings.

=> I really afraid to increase this buffer size.

Reply:
Yeah, that was not the best solution. It should work better with battle-tested solution used in scala.js using OutputStreams